### DISCUSSION TITLE
segmentation fault on debian 10 with GCC and CPU AMD ryzen 7.

### DISCUSSION THREAD

**Discussion by enricoravera** [grimme-lab/xtb#429] [Q&A]
I have compiled xtb on debian 10 with GCC and CPU AMD ryzen 7. The compilation runs smoothly, but tests fail:
5 - EXE_Singlepoint (SEGFAULT)
6 - EXE_IP/EA (SEGFAULT)
7 - EXE_GFN0-xTB (SEGFAULT)
8 - EXE_GFN1-xTB (SEGFAULT)
9 - EXE_GFN0-xTB/GBSA (SEGFAULT)
and segmentation fault is returned in any run. The compilation and execution work smoothly on an older system with the same OS and compiler but Intel CPU.
Many thanks in advance, 

Enrico

**Comment by awvwgk** [comment]
Did you set `ulimit -s unlimted`? This can even cause issues in the testsuite but cannot be enforced automatically by the build system.

**Reply by enricoravera** [reply]
Yes, I followed all the indications about setup, including changing the omp stack etc...

**Reply by awvwgk** [reply]
Strange, considered that the error happens as segmentation fault when invoking the executable but not in the test runners. Not sure what to make from this without further details.

**Reply by enricoravera** [reply]
This is the output I get after restarting the machine, setting ulimit -s unlimited, the stack size to 1G and the number of threads to half the number of actual cores. Please let me know the details you need.

Best, 

E.
```
           -------------------------------------------------
          |                Calculation Setup                |
           -------------------------------------------------

          program call               : xtb inp.xyz
          coordinate file            : inp.xyz
          number of atoms            :                     4
          number of electrons        :                    10
          charge                     :                     0
          spin                       :                   0.0
          first test random number   :      0.89184043780966

   ID    Z sym.   atoms
    1    1 H      1, 4
    2    6 C      2, 3

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x1487b3f368b0 in ???
#1  0x1487b3f35ae3 in ???
#2  0x1487b3bae83f in ???
	at /build/glibc-vjB4T1/glibc-2.28/signal/../sysdeps/unix/sysv/linux/x86_64/sigaction.c:0
#3  0x1487b3cd3820 in ???
	at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:384
#4  0x55a34a7203ef in ???
#5  0x55a34a7b228c in ???
#6  0x55a34a6fdc2e in ???
#7  0x55a34a6fe25a in ???
#8  0x55a34a6a1d5c in ???
#9  0x55a34a69cdfc in ???
#10  0x1487b3b9b09a in __libc_start_main
	at ../csu/libc-start.c:308
#11  0x55a34a69ce59 in ???
#12  0xffffffffffffffff in ???
Segmentation fault
```

**Reply by awvwgk** [reply]
From the output it might be around this place in the main code:

https://github.com/grimme-lab/xtb/blob/ecd51be214147b3f5a324e387ab83096a14d79b0/src/prog/main.F90#L411

Since I can't reproduce it, I can't really debug it. We are running `xtb` on AMD CPUs here as well and didn't notice any issues so far.
